Surface all modules on search and allow "one-way" activation.

dereksmart opened the pull request:
Implements @MichaelArestad "one-way activation" suggestion in #6788 

This PR adds a new component, `<SearchableModules />`, which has a hard-coded list of modules available for search.  It's hard-coded, because search already inherits the cards from existing tabs, and has specific settings configs.  These features will all be surfaced with the same card style for each, in order to simplify this code.  

The card is populated with the Name/descriptions from the module headers themselves.  We can change the copy of them, if so desired. 

If the module is not active, it will show a banner like this: 
![screen shot 2017-05-15 at 8 56 05 am](https://cloud.githubusercontent.com/assets/7129409/26058569/69898b9a-394c-11e7-8ecd-146f18314ea0.png)

If the module is active, we'll continue to show the card with the learn more link:
![screen shot 2017-05-15 at 8 56 18 am](https://cloud.githubusercontent.com/assets/7129409/26058576/72a589cc-394c-11e7-9856-6e53c99edfef.png)

Please test the following (hard-coded) modules that will now appear in search: 
- 'contact-form',
- 'custom-css',
- 'enhanced-distribution',
- 'json-api',
- 'latex',
- 'notes',
- 'omnisearch',
- 'shortcodes',
- 'shortlinks',
- 'widget-visibility',
- 'widgets'

Please make sure that you're not able to search these as a non-admin. 

Please make sure that they show a disabled card if in dev mode, and the feature requires a connection, JSON API.

jeherve commented on _inc/client/searchable-modules/index.jsx:
```diff
@@ -0,0 +1,135 @@
+			'enhanced-distribution',
+			'json-api',
+			'latex',
+			'notes',
```
Should Monitor be on that list as well, as folks can activate it when it's been deactivated?

dereksmart commented:
@jeherve good calls.  I modified this to use the same actions as the settings page.  We should be working to deprecate the old `activateModule` actions anyway.  You should now see a consistent experience re: the notices.  

Also added monitor to the list.

eliorivero commented:
Works fine, merging.
